packages/ui5-task-pwa-enabler/lib/pwaEnabler.js

```diff
@@ -0,0 +1,286 @@
+        // Remove prefix
+        let path = resources[i].getPath().replace(rootDir, "");
+        preCache = preCache + '\n"' + path + '",';
+    }
+    // Concatenate all regular expressions for networkFirst
+    let networkFirst = "";
+    for (let i = 0; i < parameters.networkFirst.length; i++) {
+        networkFirst = networkFirst + '\n"' + parameters.networkFirst[i] + '",';
+    }
+    // Concatenate all regular expressions for networkFirst
+    let avoidCaching = "";
+    for (let i = 0; i < parameters.avoidCaching.length; i++) {
+        avoidCaching = avoidCaching + '\n"' + parameters.avoidCaching[i] + '",';
+    }
+    // Write the service worker
+    await renderServiceWorker(path.join(__dirname, "../templates/advancedCache.js"), {
+        offlinePage: parameters.offlinePage,
+        preCache: preCache,
+        networkFirst: networkFirst,
+        avoidCaching: avoidCaching
+    });
+    return undefined;
+}
+
+/**
+ * Utility function to easily write the service worker
+ *
+ * @param {string} serviceWorkerTemplate path to the service worker template file
+ * @param {object} view will be passed to mustache.render to read values from it
+ * @returns {Promise<undefined>}
+ */
+async function renderServiceWorker(serviceWorkerTemplate, view) {
+    let path = rootDir + "/service-worker.js";
+    let swCode = fs.readFileSync(serviceWorkerTemplate);
+    view.timestamp = new Date().getTime().toString();
+    swCode = mustache.render(swCode.toString(), view);
+    await writeFile({path: path, content: swCode});
+    return undefined;
+}
+
+/**
+ * This function appends the given string representation of an html element to the <head> element of the
+ * index.html
+ *
+ * @param {string} element The string representation of the element that should be injected
+ * @returns {Promise} resolving once the data has been written
+ */
+async function appendToIndexHtmlHead(element) {
+    let resource = await ws.byPath(rootDir + "/index.html");
+    let indexHtmlContent = await resource.getString();
+    let doc = new JSDOM(indexHtmlContent);
+    // Get the head node and append the parsed element:
+    //                       <html>            <head>
+    doc.window.document.firstElementChild.firstElementChild.appendChild(JSDOM.fragment(element));
+    indexHtmlContent = doc.serialize();
+    resource.setString(indexHtmlContent);
+    await ws.write(resource);
+    return undefined;
+}
+
+/**
+ * Merges default_manifest with given manifest from your-sapui5-app/ui5.yaml
+ * No nested merge e.g. if type in icons is missing it will not be filled with default type.
+ * But if icons as a whole is missing it will be filled with default icons.
+ * And adds manifest-link to index.html
+ *
+ * @param {object} manifestConfig Every parameter supplied will override the corresponding default value.
+ */
+async function addManifest(manifestConfig) {
+    let manifest = Object.assign(default_manifest, manifestConfig); //default_manifest will be overidden
```

Comment:
or create a complete copy of all to not modify the default manifest - in your case this is no issue as the task is not reuse - but this would be more future safe, prefer to use:

`Object.assign({}, default_manifest, manifestConfig)`